test-harness/src/test/java/com/netflix/conductor/tests/integration/End2EndGrpcTests.java

```diff
@@ -190,40 +190,117 @@ public void testAll() throws Exception {
         assertEquals(1, wf.getTasks().size());
     }
+
+    @Test
+    public void testEphemeralWorkflowsWithStoredTasks() throws Exception {
+        List<TaskDef> definitions = createAndRegisterTaskDefinitions("storedTaskDef", 5);
+
+        WorkflowDef workflowDefinition = new WorkflowDef();
+        workflowDefinition.setName("testEphemeralWorkflow");
+
+        WorkflowTask workflowTask1 = createWorkflowTask("storedTaskDef1");
+        WorkflowTask workflowTask2 = createWorkflowTask("storedTaskDef2");
+
+        workflowDefinition.getTasks().add(workflowTask1);
+        workflowDefinition.getTasks().add(workflowTask2);
+
+        String workflowExecutionName = "ephemeralWorkflow";
+        StartWorkflowRequest workflowRequest = new StartWorkflowRequest()
+                .withName(workflowExecutionName)
+                .withWorkflowDef(workflowDefinition);
+
+        String workflowId = workflowClient.startWorkflow(workflowRequest);
+        assertNotNull(workflowId);
+
+        Workflow workflow = workflowClient.getWorkflow(workflowId, true);
+        WorkflowDef ephemeralWorkflow = workflow.getWorkflowDefinition();
+        assertNotNull(ephemeralWorkflow);
+        assertEquals(workflowDefinition.getName(), ephemeralWorkflow.getName());
+    }
+
+    @Test
+    public void testEphemeralWorkflowsWithEphemeralTasks() throws Exception {
+        WorkflowDef workflowDefinition = new WorkflowDef();
+        workflowDefinition.setName("testEphemeralWorkflowWithEphemeralTasks");
+
+        WorkflowTask workflowTask1 = createWorkflowTask("ephemeralTask1");
+        TaskDef taskDefinition1 = createTaskDefinition("ephemeralTaskDef1");
+        workflowTask1.setTaskDefinition(taskDefinition1);
+
+        WorkflowTask workflowTask2 = createWorkflowTask("ephemeralTask2");
+        TaskDef taskDefinition2 = createTaskDefinition("ephemeralTaskDef2");
+        workflowTask2.setTaskDefinition(taskDefinition2);
+
+        workflowDefinition.getTasks().add(workflowTask1);
+        workflowDefinition.getTasks().add(workflowTask2);
+
+        String workflowExecutionName = "ephemeralWorkflowWithEphemeralTasks";
+        StartWorkflowRequest workflowRequest = new StartWorkflowRequest()
+                .withName(workflowExecutionName)
+                .withWorkflowDef(workflowDefinition);
+
+        String workflowId = workflowClient.startWorkflow(workflowRequest);
+        assertNotNull(workflowId);
+
+        Workflow workflow = workflowClient.getWorkflow(workflowId, true);
+        WorkflowDef ephemeralWorkflow = workflow.getWorkflowDefinition();
+        assertNotNull(ephemeralWorkflow);
+        assertEquals(workflowDefinition.getName(), ephemeralWorkflow.getName());
+
+        List<WorkflowTask> ephemeralTasks = ephemeralWorkflow.getTasks();
+        assertEquals(2, ephemeralTasks.size());
+        for (WorkflowTask ephemeralTask : ephemeralTasks) {
+            assertNotNull(ephemeralTask.getTaskDefinition());
+        }
+
+    }
+
```

Comment:
**Concern:** Can we also add a test that is using a mix of ephemeral and stored task definitions?